src/server/cluster/cluster_family.cc

```diff
@@ -718,12 +723,79 @@ void ClusterFamily::MigrationConf(CmdArgList args, ConnectionContext* cntx) {
     }
   }
 
-  cntx->conn()->SetName("slot_migration_ctrl");
+  auto sync_id = CreateMigrationSession(cntx, port);
 
-  cntx->SendLong(shard_set->size());
+  cntx->conn()->SetName("slot_migration_ctrl");
+  auto* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
+  rb->StartArray(2);
+  rb->SendLong(sync_id);
+  rb->SendLong(shard_set->size());
   return;
 }
 
+uint32_t ClusterFamily::CreateMigrationSession(ConnectionContext* cntx, uint16_t port) {
+  std::lock_guard lk(migration_mu_);
+  auto sync_id = next_sync_id_++;
+  auto info = make_shared<MigrationInfo>(shard_set->size(), cntx->conn()->RemoteEndpointAddress(),
+                                         sync_id, port);
+  auto [it, inserted] = migration_infos_.emplace(sync_id, info);
+  CHECK(inserted);
+  return sync_id;
+}
+
+void ClusterFamily::Flow(CmdArgList args, ConnectionContext* cntx) {
+  CmdArgParser parser{args};
+  auto [sync_id, shard_id] = parser.Next<uint32_t, uint32_t>();
+
+  if (auto err = parser.Error(); err) {
+    return cntx->SendError(err->MakeReply());
+  }
+
+  VLOG(1) << "Create flow "
+          << " sync_id: " << sync_id << " shard_id: " << shard_id << " shard";
+
+  cntx->conn()->SetName(absl::StrCat("migration_flow_", sync_id));
+
+  auto info = GetMigrationInfo(sync_id);
+  if (!info)
+    cntx->SendError(kIdNotFound);
+
+  info->flows[shard_id].conn = cntx->conn();
+
+  cntx->conn()->Migrate(shard_set->pool()->at(shard_id));
+
+  cntx->SendOk();
+}
+
+void ClusterFamily::Sync(CmdArgList args, ConnectionContext* cntx) {
```

Comment:
Why do we need the sync sub command? cant this be part of the flow sub command?

Reply:
Do you mean to start sync after getting the flow command and don't wait for all flow commands?

Reply:
right. I am asking.. dont know what is there reason for this. I know we have this in replication but not sure what is the reason for this in replication